### PR TITLE
Issue #2742: Move 'Add role' form to new page.

### LINKED ISSUE #2742: [D8][UX] Replace the 'Add role' button with a '+ Add role' action link, and move the `Role name` form to its own page.
## Describe your issue or idea

Throughout the rest of the UI we have changed everything to "+ Add xyz" links, but in the "Roles" page we use a different pattern.

### Steps to reproduce (if reporting a bug)

Navigate to `admin/config/people/roles`

### Actual behavior (if reporting a bug)

image

### Expected behavior (if reporting a bug)

image

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/2742

### CODE REVIEW

File: core/modules/user/user.module [hunk 1]
```diff
@@ -1442,6 +1442,16 @@ function user_menu() {
     'access arguments' => array('administer permissions'),
     'file' => 'user.admin.inc',
   );
+  $items['admin/config/people/roles/add'] = array(
+    'title' => 'Add role',
+    'description' => 'Add a new user role.',
+    // 'page callback' => 'backdrop_get_form',
```
Comment (docwilmot) [inline core/modules/user/user.module]: Comment to delete. Same on line 1452.

File: core/modules/user/user.module [hunk 2]
```diff
@@ -1442,6 +1442,16 @@ function user_menu() {
     'access arguments' => array('administer permissions'),
     'file' => 'user.admin.inc',
   );
+  $items['admin/config/people/roles/add'] = array(
+    'title' => 'Add role',
```
Comment (docwilmot) [inline core/modules/user/user.module:1447]: The page title doesnt seem to inherit this title still just says "Roles"

Comment (ghost) [reply core/modules/user/user.module:1447]: Hmm true. Not sure why that is. Let's discuss in the issue.

**Review by klonos — APPROVED** [review]
LGTM 👍

**Comment by quicksketch** [comment]
Merged into 1.x for 1.19.0. Thanks!

**Pull request merged**